### DISCUSSION TITLE
Problem in "transformationMatrix" When using "geometry-templates"

### DISCUSSION THREAD

**Discussion by HibaDoi** [cityjson/specs#201] [General]
I'm struggling with a problem while working with CityJSON. I'm using a geometry template, but I encountered a very unusual issue. I tried using the following template( the same template in [here:](https://www.cityjson.org/dev/geom-templates/):
{
  "type": "CityJSON",
  "version": "1.1",
  "CityObjects": {
    "mybigtree": {
      "type": "SolitaryVegetationObject",
      "attributes": {
        "size": "very big",
        "shape": "kind of a box"
      },
      "geometry": [
        {
          "type": "GeometryInstance",
          "template": 0,
          "boundaries": [0],
          "transformationMatrix": [1,0,0,0,0,1,0,0,0,0,1,0,0,0,0,1]
        }
      ]
    }
  },
  "vertices": [[174000,875000,0]],
  "geometry-templates": {
    "vertices-templates": [
      [0,0,5],[10,0,5],[10,10,5],[0,10,5],[0,0,15],[10,0,15],[10,10,15],[0,10,15]
    ],
    "templates": [
      {
        "boundaries": [[[0,3,2,1]],[[4,5,6,7]],[[0,1,5,4]],[[1,2,6,5]],[[2,3,7,6]],[[0,4,7,3]]],
        "lod": "1",
        "type": "MultiSurface"
      }
    ]
  },
  "transform": {
    "scale": [0.001,0.001,0.001],
    "translate": [85000,446000,0]
  },
  "metadata": {
    "referenceSystem": "https://www.opengis.net/def/crs/EPSG/0/7415"
  }
}

However, when I change the transformationMatrix from
"transformationMatrix": [1,0,0,0,0,1,0,0,0,0,1,0,0,0,0,1]
to:
"transformationMatrix": [0,1,0,0,-1,0,0,0,0,0,1,0,0,0,0,1]

the entire position of the object changes, which seems absurd because I’m only applying a rotation.

i display the cityjson in piero [here](https://piero.giro3d.org/?tour=none)

**Comment by hugoledoux** [comment]
but the transformation matrix is not only doing a rotation, see the matrices there: https://www.cityjson.org/dev/geom-templates/

**Reply by HibaDoi** [reply]
yess, i applied here only the rotation part i did not add translation as you see the last column is 0 0 0 1.

**Reply by HibaDoi** [reply]
![image](https://github.com/user-attachments/assets/5643c660-f496-41c8-8036-a35aeed82e57)
i just applied a 90 ° rotation around the Z axis .

**Comment by clausnagel** [comment]
The points are rotated about the origin (0,0,0). Try and move the origin to the center of your template geometry (at least on the xy-plane if you want to rotate around the z-axis).

**Reply by HibaDoi** [reply]
I hanged the vertices-templates to that:"geometry-templates"
:{"vertices-templates":[[ -5.00 ,  -5.00 , -5],
[  5.0 ,  -5.00 , -5],
[  5.0 ,   5.0 , -5],
[ -5.00 ,   5.0 , -5],
[ -5.00 ,  -5.00 ,  5],
[  5.0 ,  -5.00 ,  5],
[  5.0 ,   5.0 ,  5],
[ -5.00 ,   5.0 ,  5]]

the center of this template is 0 0 0 but i have the same problem with the coordinate.

**Reply by clausnagel** [reply]
Maybe a software issue of your viewer? Works for me using the KITModelViewer. I used a 45° rotation for visibility reasons. 

![image](https://github.com/user-attachments/assets/24e31817-dffe-4521-920a-09731d8c4cbf)

**Reply by HibaDoi** [reply]
Intersting ,I think the problem is indeed in the viewer
